### ISSUE TITLE
Menu - Visual treatment for group styling

### ISSUE THREAD

**Issue by KennyAtHPE** [grommet/hpe-design-system#2907]
Referring to the feedback below, make a design decision and then reflect that in Figma and grommet.

## Suggestion
- hold a meeting with the team to discuss and make a decision

## Deliverables
- [ ] any updates to the Figma component
- [ ] any updates to grommet or the hpe theme

[Question](https://www.figma.com/file/Lh01flLH73cFNGZiFxEFth/HPE---Office-Hours?node-id=1542%3A16295) came up in Office Hours (9/21) on what the visual treatment should be for group styling within menus. 

![image](https://user-images.githubusercontent.com/86075038/191564325-f1730e5d-1589-4c0d-90f4-5637c9283f1a.png)

Comments Sam G. made on this issue:

I feel very strongly that the 1st option (far left) is the best for a number of reasons:
- In a menu, you should always have the hover apply to a valid option — not the menu background itself. If you have additional negative space above and/or below the horizontal separator, this is inevitably what you end up with (hovering over the menu background, not a functional menu option). As a result, when you move the cursor across the boundaries from one menu option to the separator and then over another menu option, it isn’t a smooth transition & looks (& feels) “messy” and unintentional. This is something a static Figma file doesn’t convey, but actual UIs do quite well.
- From a graphic design standpoint, it is important to avoid multiple alternating lines that go from light-dark-light-dark or visa-versa. The result can be visual “vibration”, which can be disconcerting to many users. There is an entire field of “optical art” or “op art” that makes use of this characteristic of the human visual system. Spend a little time googling “op art”, and you’ll see why it’s important to avoid this.
- If you eliminate the negative space above & below the horizontal separator, imagine what the third (right-most) option would look like: the background + border would look “messy” and unintentional by comparison. Horizontal separators going edge-to-edge would look far cleaner by comparison.

**Comment by vavalos5** [comment]
i did a bit of investigating on the styling we have on the design system site, Figma and Grommet. It looks like on DS site, we have inconsistencies where we have extra padding top and bottom of the menu component and some areas where we don't. Grommet doesn't have any and Figma styles the component with 6px top and bottom.

When I took a look at our select component, that one doesn't have any top and bottom padding. So, I would agree with Sam on this case that the hover should be applied to eliminate the negative space above and below the separators.

**Comment by halocline** [comment]
Background for context:

- https://github.com/grommet/hpe-design-system/issues/2621
- https://github.com/grommet/hpe-design-system/pull/2636

**Comment by vavalos5** [comment]
Hmm.. Second guessing my previous decision. Upon looking at how Chrome and GitHub handle menus, there seems to be padding top and bottom of grouped items. These help provide some space between grouped items.
<img width="631" alt="Screen Shot 2022-10-10 at 3 35 13 PM" src="https://user-images.githubusercontent.com/62117454/194962273-fe1c9b13-c70f-497c-bbbb-8127212893cc.png">
<img width="613" alt="Screen Shot 2022-10-10 at 3 34 50 PM" src="https://user-images.githubusercontent.com/62117454/194962274-cb9073df-41f6-4118-b9a7-01d297d833ea.png">
<img width="490" alt="Screen Shot 2022-10-10 at 3 33 56 PM" src="https://user-images.githubusercontent.com/62117454/194962275-2835d5e4-48da-4f36-b8d6-0f4dd7bc79b3.png">
<img width="217" alt="Screen Shot 2022-10-10 at 3 31 55 PM" src="https://user-images.githubusercontent.com/62117454/194962276-e1dade75-f29f-40b4-8fd6-1c14a5ce944e.png">

**Comment by vavalos5** [comment]
Sharing with Designers tomorrow to collect feedback. [Figma file](https://www.figma.com/file/DyFWlzLBUtK5KH5jJhwoCB/HPE-Menu-Component?node-id=157%3A0).

<img width="1097" alt="Screen Shot 2022-10-10 at 4 38 48 PM" src="https://user-images.githubusercontent.com/62117454/194967954-c3932213-bafe-434f-81a8-7d14de3f517e.png">

**Comment by vavalos5** [comment]
I've reviewed comments left by Matt and reviewed with Kenny. We've decided on Option 1. Ready to hand off to @halocline 

![Image](https://user-images.githubusercontent.com/62117454/195206196-6fb9b807-f665-469f-ac2a-0d4c514782c0.png)


Another note for me to come back to later is that within Figma, the ungrouped items have 4px top and bottom. I will align to the design system site where there is no padding top and bottom.
